frontend/src/components/BalanceCards.jsx

```diff
@@ -15,35 +15,61 @@ const BalanceCards = ({ walletId }) => {
   ]);
 
   const isMobile = useMatchMedia('(max-width: 768px)');
+  const [activeIndex, setActiveIndex] = useState(0);
+  const scrollRef = useRef(null);
+  
+  const handleScroll = () => {
+    if (scrollRef.current) {
+      const { scrollLeft, scrollWidth, clientWidth } = scrollRef.current;
+      const scrollPercentage = scrollLeft / (scrollWidth - clientWidth);
+      const index = Math.round(scrollPercentage);
+      setActiveIndex(index);
+    }
+  };
+
 
   useEffect(() => {
     getBalances(walletId, setBalances);
   }, [walletId]);
 
+  useEffect(() => {
+    const scrollElement = scrollRef.current;
+    if (scrollElement) {
+      scrollElement.addEventListener("scroll", handleScroll);
+      return () => scrollElement.removeEventListener("scroll", handleScroll);
+    }
+  }, []);
+
   return (
-    <div className="balance-container">
+    <div className='balance-card'>
+    <div className="balance-container" ref={scrollRef}>
       {balances.map((balance) =>
         isMobile ? (
           <div className="balance-item" key={balance.title}>
-            <label htmlFor="icon" className="balance-title">
-              {balance.icon}
-            </label>
             <div className="title-container">
-              <label htmlFor={balance.title}>{balance.title} Balance:</label>
-              <label htmlFor={balance.title}>{balance.balance}</label>
+              <label htmlFor="icon" className="balance-title">
+              <span className="token-icon">{balance.icon}</span>
+              </label>
+              <label htmlFor={balance.title}>{balance.title} Balance</label>
             </div>
+              <label htmlFor={balance.title}>{balance.balance}</label>
           </div>
         ) : (
           <div className="balance-item" key={balance.title}>
             <label htmlFor={balance.title} className={'balance-title'}>
-              {balance.icon}
-              {balance.title} Balance:
+            <span className="token-icon">{balance.icon}</span>
+              {balance.title} Balance
             </label>
             <label htmlFor={balance.title}>{balance.balance}</label>
           </div>
         )
       )}
     </div>
+      <div className="pagination">
+        <div className={`dot ${activeIndex === 0 ? "active" : ""}`}></div>
+        <div className={`dot ${activeIndex === 1 ? "active" : ""}`}></div>
+      </div>
+      </div>
```

Comment:
create a separate component for this, number of dots should be dynamic, though, based on the number of balances, because in the future there can be more than 4 balances, so it should be calculated, like dot per each 2 balances.

Also, each dot should be clickable and scroll horizontally to the left or right

Reply:
okay, on it